### ISSUE TITLE
[Javascript] TSX Highlighting Problem When Using Ternary and Template Literal

### ISSUE THREAD

**Issue by QuestionMark3** [sublimehq/Packages#3903]
### What happened?

When creating a react component in TSX with:
- a ternary render
- a template literal in one of the conditions

the syntax highlights incorrectly:

<img width="748" alt="Screenshot 2024-01-12 at 5 12 28 PM" src="https://github.com/sublimehq/Packages/assets/32991975/bc210200-7bfc-4daa-ab04-9bf556b20bb1">

When I remove the expressions from the template literal it seems to work fine:

<img width="705" alt="Screenshot 2024-01-12 at 5 17 52 PM" src="https://github.com/sublimehq/Packages/assets/32991975/aa141f8c-e6a6-4107-9ae9-9c0c7b0fefe5">

There seems to be no issues when working with JSX.

[code for convenience](https://github.com/sublimehq/Packages/files/13924632/sublime-tsx-syntax-highlight-bug.tsx.zip)

**Comment by Thom1729** [comment]
Shorter example:

```tsx
a ? () : (
  <a>
    <a
      a={() => `${a}`}
    />
  </a>
)
```

I think what's happening is that it tries to parse the initial parens as the beginning of an arrow function, with the colon beginning a return type annotation. For whatever reason it never hits a token that makes it give up on that interpretation. The example is extremely brittle and even minor changes seem to fix it. It might take me a bit of figuring to determine why exactly it's making it that far without failing the arrow function .

**Comment by Thom1729** [comment]
Well, I finally got to look at this in detail and it looks like it works in the latest version of the syntax. If so, it should work in the next Sublime release.

**Comment by deathaxe** [comment]
Fixed by #3863
